content/doc/developer/plugin-development/incrementals.adoc

```diff
@@ -0,0 +1,108 @@
+Also, it allows keeping track and switching between incremental versions.
+Not only that, it allows others to also not have to build Jenkins locally to try out the changes or collaborate with
+you on your awesome project!
+
+Sounds cool? Let's see how to go about it.
+
+== 1. Create a Pull Request in Jenkins core
+
+First step is to link:https://github.com/jenkinsci/jenkins/pulls[create a PR on github] by clicking the "New Pull
+Request" button and push the changes to Jenkins core.
+
+image::/images/developer/plugin-development/incrementals/create-pr.png[]
+
+Note: Make sure that you rebase your branch with the latest changes from master, otherwise incrementals version
+will NOT be created.
+
+== 2. Wait for all checks to pass
+
+Wait for 'continuous-integration/jenkins' to perform checks, this can take about two hours.
+
+== 3. Obtain the Incrementals version
+
+After the checks are completed, you should get a message like below:
+
+image::/images/developer/plugin-development/incrementals/checks.png[]
+
+Click on 'Details' adjacent to 'continuous-integration/jenkins/incrementals — Deployed to Incrementals.'
+
+Note: If you only see one check 'continuous-integration/jenkins/pr-merge — This commit looks good' then there is an
+error due to which incrementals version was not created. One of the most likely reasons is that latest changes from
+master have not been pulled into the branch. You can click on 'Details' adjacent to it to investigate this error from
+the logs.
+
+== 4. Copy the Version ID
+
+You should be redirected to a page like below:
+
+image::/images/developer/plugin-development/incrementals/incrementals.png[]
+
+Copy the version ID, which in this example would be `2.238-rc29961.5c3c5871cca6`.
+
+== 5. Update the plugin POM
+
+There are two changes you want to make to your plugin POM.
+
+First, update the Jenkins version to the version ID copied in the previous step:
+
+```
+...
+    <properties>
+        <jenkins.version>2.238-rc29961.5c3c5871cca6</jenkins.version>
+        ...
+    </properties>
+...
+```
+
+Then, add the incrementals repository to the list of repositories:
+
+```
+...
+    <repositories>
+        ...
+        <repository>
+            <id>incrementals.jenkins-ci.org</id>
+            <url>https://repo.jenkins-ci.org/incrementals/</url>
+        </repository>
+        ...
+    </repositories>
+...
+```
+
+Now you can run your project like you normally do (say using `hpi:run`), and Jenkins will it automatically run it
+against the incrementals version of Jenkins core you specified.
+
+== 6. (Optional) Enable usage of Beta API
+
+If the API you are building or are going to try has been marked
+link:https://www.jenkins.io/blog/2018/04/30/using-the-beta-annotation/[beta], then you would also have to add the
+beta tag, like so:
+
+```
+...
+    <properties>
+        <jenkins.version>2.238-rc29961.5c3c5871cca6</jenkins.version>
+        <useBeta>true</useBeta>
+        ...
+    </properties>
+...
+```
```

Comment:
I would drop this section. Not related to Incrementals at all, and not great advice to offer to the world—the idea is that only someone _in contact with_ the author of the beta API (perhaps the very same person) should be calling it, in which case you would be told to use this flag in a dedicated message.

```suggestion
```